frontend/src/layout/lemonade/RedesignOptIn.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { useState } from 'react'
+import posthog from 'posthog-js'
+import { featureFlagLogic } from '../../lib/logic/featureFlagLogic'
+import { FEATURE_FLAGS } from '../../lib/constants'
+import { useValues } from 'kea'
+import './RedesignOptIn.scss'
+import { preflightLogic } from '../../scenes/PreflightCheck/logic'
+import { LemonSwitch } from '../../lib/components/LemonSwitch/LemonSwitch'
+
+export function RedesignOptIn(): JSX.Element | null {
+    const { featureFlags } = useValues(featureFlagLogic)
+    const { preflight } = useValues(preflightLogic)
+
+    const [isSwitchChecked, setIsSwitchChecked] = useState(featureFlags[FEATURE_FLAGS.LEMONADE] as boolean)
+    const [isSwitchLoading, setIsSwitchLoading] = useState(false)
+
+    return preflight?.cloud || preflight?.is_debug ? (
+        <span className="RedesignOptIn">
+            <label htmlFor="redesign-opt-in" className="info">
+                {featureFlags[FEATURE_FLAGS.LEMONADE]
+                    ? "You're using the new design"
+                    : "Try PostHog's navigation redesign!"}
+            </label>
+            <LemonSwitch
+                id="redesign-opt-in"
+                onChange={(checked) => {
+                    setIsSwitchLoading(true)
+                    setIsSwitchChecked(checked)
+                    posthog.people.set('opted_into_lemonade', checked, () =>
```

Comment:
Late posthumous review, we should've added an event too to report when users opt-in/out as we don't have properties over time we won't know for instance if a user enabled it, got scared and went back. Autocapture will also not be perfect as we won't know for sure the value of the switch when clicked.

Reply:
Don't we already do this with the `posthog.capture(checked ? `opted into navigation redesign` : 'opted out of navigation redesign')` line below? I've been using that data